runner/src/internal.rs

```diff
@@ -99,10 +101,33 @@ pub struct BuildClient {
     pub client_rust_target: Option<String>,
 }
 
+#[derive(Clone, Debug)]
+pub enum ServerVariant {
+    Base,
+    Logless,
+    NoIntrospectionServer,
+}
+
+impl std::str::FromStr for ServerVariant {
```

Comment:
FYI this may make things more concise: https://docs.rs/enum_derive/0.1.7/enum_derive/macro.EnumFromStr.html (I haven't tried it myself but I'm curious if it works well)

Additionally, impl'ing iter via https://docs.rs/enum_derive/0.1.7/enum_derive/macro.IterVariants.html may also make it easier to have steps that build all the variants automatically by mapping over the variants.

Reply:
I'm going to skip this &ndash; partly because it's an extra pair of deps, but mostly because `EnumFromStr` "requires an exact match of the variant name" (instead of the existing lowercase option names).